src/crypto/into_fr.rs

```diff
@@ -0,0 +1,55 @@
+use pairing::bls12_381::Fr;
+use pairing::{Field, PrimeField};
+
+/// A conversion into an element of the field `Fr`.
+pub trait IntoFr: Copy {
```

Comment:
Is there an advantage of this specialized type over the more idiomatic `Into<Fr>`? I think `Fr::into` would look better.

Reply:
I'm not allowed to add `impl X for Y` in a crate to which neither `X` nor `Y` belongs. (With generic arguments it's a bit more complicated, I think, but I can't e.g. `impl Into<Fr> for usize`.)